include/oneapi/dpl/internal/dynamic_selection_impl/backend_traits.h

```diff
@@ -0,0 +1,55 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_INTERNAL_BACKEND_TRAITS_H
+#define _ONEDPL_INTERNAL_BACKEND_TRAITS_H
+
+#include <utility>
+#include <cstdint>
+#include <type_traits>
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace experimental
+{
+namespace internal
+{
+    template <typename Backend>
+    auto
+    has_lazy_report_impl(...) -> std::false_type;
+
+    template <typename Backend>
+    auto
+    has_lazy_report_impl(int) -> decltype(std::declval<Backend>().lazy_report(), std::true_type{});
+
+    template <typename Backend>
+    struct has_lazy_report : decltype(has_lazy_report_impl<Backend>(0))
+    {
+    };
+
+}//namespace internal
+
+namespace backend_traits {
+    template <typename S>
+    struct lazy_report_value
+    {
+        static constexpr bool value = ::oneapi::dpl::experimental::internal::has_lazy_report<S>::value;
+    };
+    template <typename S>
+    inline constexpr bool lazy_report_v = lazy_report_value<S>::value;
```

Comment:
@vossmjp should `backend_traits`, `lazy_report`, `lazy_report_v` etc. names be considered a (future) public API of DS backends, or there is no such intent?
I am asking because `backend_traits` is put into `oneapi::dpl[::experimental]` and looks more like a potential public name. Still experimental and so OK to change; nevertheless, should we have a naming discussion for those?

Reply:
`backend_traits `has traits that are meant at the backend level, not the policy level like rest of the traits. The reason for creating another namespace inside `oneapi::dpl::experimental` was to differentiate between those two and since the backend_traits.h file is inside the include/oneapi/dpl/internal/dynamic_selection_impl folder it is assumed that it is an implementation detail. 
The backend is never exposed to the user and thus `backend_traits`,` lazy_report`,` lazy_report_v` etc. names may not be  a public API of DS backends. 
Let us know your opinions, @vossmjp

Reply:
@AnuyaWelling2801 and @akukanov, it is expected that we will eventually document how to create custom backends. This will be a feature for policy writers and not end users of dynamic selection. So, it will become a documented set of APIs but for a much more limited audience. Previously, there was only an undocumented contract between the current concrete policies and the single current concrete backend. What we might do is include the internal/dynamic_selection_traits.h (which includes policy_traits.h) in the dynamic_selection header but not the backend_traits.h.